src/suites/cts/validation/error_scope.spec.ts

```diff
@@ -0,0 +1,128 @@
+export const description = `
+error scope validation tests.
+`;
+
+import { getGPU } from '../../../framework/gpu/implementation.js';
+import { Fixture, TestGroup } from '../../../framework/index.js';
+
+class F extends Fixture {
+  device: GPUDevice = undefined!;
+
+  async init(): Promise<void> {
+    super.init();
+    const gpu = getGPU();
+    const adapter = await gpu.requestAdapter();
+    this.device = await adapter.requestDevice();
+  }
+
+  createErrorBuffer(): void {
+    this.device.createBuffer({
+      size: 1024,
+      usage: 0xffff, // Invalid GPUBufferUsage
+    });
+    // TODO: Remove when chrome does it automatically.
+    this.device.getQueue().submit([]);
+  }
+
+  async expectUncapturedError(fn: Function): Promise<void> {
+    // TODO: Make arbitrary timeout value a test runner variable
+    const TIMEOUT_IN_MS = 1000;
+
+    return this.asyncExpectation(async () => {
+      try {
+        const promise = new Promise((resolve, reject) => {
+          const timeoutId = window.setTimeout(() => {
```

Comment:
nit: shouldn't use `window.`, since we may want this to work on workers too

Reply:
Done